src/platform/Tizen/ThreadStackManagerImpl.h

```diff
@@ -57,6 +57,11 @@ class ThreadStackManagerImpl : public ThreadStackManager
     bool _TryLockThreadStack() { return false; }            // Intentionally left blank
     void _UnlockThreadStack() {}                            // Intentionally left blank
 
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT
```

Comment:
I'm not sure whether I read this correctly, but are these methods required only in case where ThreadManager is based on `GenericThreadStackManagerImpl_OpenThread.hpp`? If that's right, Tizen is not based on this generic implementation.

Also, I would suggest to clearly document in what cases these functions are required. Maybe even do not define them in the `platform/ThreadStackManager.h` because wait and notify is not part of `ThreadStackManager` but only a specific implementation of `_ClearAllSrpHostAndServices()`.

Reply:
I can add documentation. However, since it is required in the `GenericThreadStackManagerImpl`, the synchronization mechanism needs to be implementable for each platform in their `ThreadManager` as each platform can have different apis or mechanisms as you can see with the implementation.

The release synchronization is needed. is because we are sending the request to the OTBR (srp server) to clear all its SRP hosts and services with the current client (the device). We need to make sure the request gets to the OTBR before we reboot. We do this by waiting on the completion callback. If we don't do that. Chance are you will reboot and the OTBR will not have received the request and will not have cleared its srp services. I tested and can confirm the behaviour.

Now I don't know your platform thread APIs and haven't tested it on your hardware, but I would expect the same requirement. Aka you can't reboot directly after sending the clear request. You need to validate the otbr acknowledged the request.

In any way, if you don't use them leave them unimplemented just as it is done for the Tizen platform with similar functions above.
```
_LockThreadStack
_TryLockThreadStack
_UnlockThreadStack
```

Reply:
> We need to make sure the request gets to the OTBR before we reboot. We do this by waiting on the completion callback.

Yes, that's in case when the API (in this particular case) to clear SRP hosts and services is asynchronous. But not in case when the API used in the `_ClearAllSrpHostAndServices()` is already synchronous (and to be honest I do not know whether the Tizen API is implemented correctly, whether it's indeed synchronous, but it looks like so). So, it is strange that particular implementation of the `ThreadStackManager` interface requires implementation-specific methods in the interface itself. I understand that decoupling implementation and interface might be tricky in that case...